packages/stencil-library/src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,170 @@
+import { Component, h } from "@stencil/core";
+import { Event, EventEmitter, HTMLStencilElement, Listen, Method, Prop, State, Watch } from "@stencil/core/internal";
+import { Element } from '@stencil/core';
+import { RadioGroupValidityState, NjwdsChangeEventDetail, NjwdsInvalidEventDetail } from "../../interface";
+
+const validityStates = Object.freeze({
+    VALID: {
+        valid: true,
+        valueMissing: false
+    },
+    MISSING_VALUE: {
+        valid: false,
+        valueMissing: true
+    },
+})
+
+@Component({
+    tag: "njwds-radio-group",
+    styles: `
+        njwds-radio-group .nj-radio-group__validation--message {
+            font-weight: bold;
+            color: #b50909;
+            margin-top: 0.5em;
+        }
+
+        njwds-radio-group .nj-form__required--asterisk {
+            color: #d54309;
+            margin-left: 0.2em;
+        }
+    `
+})
+export class RadioGroup {
+    @Prop() name!: string;
+    @Prop() required: boolean = false;
+    @Prop() tile: boolean = false;
+    @Prop({ reflect: true, mutable: true }) value: string = "";
+    @Prop({ reflect: true, mutable: true }) showValidity: boolean;
+    @Prop() validationMessage: string = "Please select an option.";
+
+    @State() validity: RadioGroupValidityState = validityStates.VALID;
+
+    @Event() njwdsChange: EventEmitter<NjwdsChangeEventDetail>;
+    @Event() njwdsInvalid: EventEmitter<NjwdsInvalidEventDetail>;
+
+    @Element() private hostElement: HTMLStencilElement
+
+    private computeValidity(): RadioGroupValidityState {
+        if (this.required && !this.value) {
+            return validityStates.MISSING_VALUE
+        }
+        return validityStates.VALID
+    }
+
+    private changeErrorPropOnRadios(isError: boolean) {
+        const njwdsRadios = this.hostElement.querySelectorAll("njwds-radio")
+
+        njwdsRadios.forEach((radio: HTMLNjwdsRadioElement) => {
+            radio.error = isError
+        })
+    }
+
+    @Listen('change')
+    changeHandler(event: Event) {
+        if ("value" in event.target && typeof event.target.value === "string") {
+            const newValue = event.target.value
+            this.value = newValue
+            this.njwdsChange.emit({
+                value: newValue
+            })
+        }
+    }
+
+    @Watch('value')
```

Comment:
Dust/pebble: Wondering if this watchValueHandler can also have oldValue arg and if so wondering if it's worth doing a check here if newValue !== oldValue. Not sure what circumstance it might happen here, but just a thought?

Reply:
Good question! I think it's okay to exclude this check — just tested and if you programmatically set `value` to the same value, it doesn't trigger the `watchValueHandler` method. (Stencil must already be doing this check under the hood which is cool, I didn't know that)